src/basic/time-util.c

```diff
@@ -1311,6 +1313,17 @@ bool timezone_is_valid(const char *name) {
         if (!S_ISREG(st.st_mode))
                 return false;
 
+        fd = open(t, O_RDONLY|O_CLOEXEC);
+        if (fd < 0)
+                return false;
```

Comment:
hmmm, stat() followed by open() always has this racy feel to it, even though it doesn't really matter in this case. Nonetheless, could you rearrange this, so that we open the file before statting it, and then use fstat() rather that stat() to check whether it's actually a regular file?

or even better, maybe even just use the existing fd_verify_regular() helper?

also, I think we should really beef this up with debug logging. I think for new code we should ensure that when we eat up error conditions like we do in this function we should at least debug log about them.

```c
fd = open(t, O_RDONLY|O_CLOEXEC);
if (fd  < 0) {
        log_debug_errno(errno, "Failed to open timezone file '%s': %m", t);
        return false;
}

r = fd_verify_regular(fd);
if (r < 0) {
        log_debug_errno(r, "Timezone file '%s' is not  a regular file: %m", t);
        return false;
}
…
```